src/app/shared/form/builder/ds-dynamic-form-ui/models/autocomplete/dynamic-autocomplete.model.ts

```diff
@@ -0,0 +1,32 @@
+import {DsDynamicInputModel, DsDynamicInputModelConfig} from '../ds-dynamic-input.model';
+import {AUTOCOMPLETE_OFF, DynamicFormControlLayout, serializable} from '@ng-dynamic-forms/core';
+import {VocabularyOptions} from '../../../../../../core/submission/vocabularies/models/vocabulary-options.model';
+import {isEmpty} from '../../../../../empty.util';
+
+export const DYNAMIC_FORM_CONTROL_TYPE_AUTOCOMPLETE = 'AUTOCOMPLETE';
+export const AUTOCOMPLETE_COMPLEX_PREFIX = 'autocomplete_in_complex_input';
+
+export interface DynamicAutocompleteModelConfig extends DsDynamicInputModelConfig {
+  minChars?: number;
+  value?: any;
+}
+
+export class DynamicAutocompleteModel extends DsDynamicInputModel {
+
+  @serializable() minChars: number;
+  @serializable() readonly type: string = DYNAMIC_FORM_CONTROL_TYPE_AUTOCOMPLETE;
+
+  constructor(config: DynamicAutocompleteModelConfig, layout?: DynamicFormControlLayout) {
+
+    super(config, layout);
+
+    if (isEmpty(this.vocabularyOptions)) {
+      this.vocabularyOptions = new VocabularyOptions('none');
+    }
+    this.autoComplete = AUTOCOMPLETE_OFF;
+    this.minChars = config.minChars || 3;
```

Comment:
it is like magic

Reply:
done

Reply:
Add a comment why 3 is a good constant. The idea is to have some kind of description. In a sense that we think its useful to start searching for matches after the user enters at least 3 chars ....

Reply:
This conversation is outdated. I added constant there, new code: 
![image](https://user-images.githubusercontent.com/90026355/182126932-cc8e7c7a-4496-43a8-ab13-71a710dd2cdd.png)